mvvm_cards/src/views/toolbar/MainDocumentPageToolbar.tsx

```diff
@@ -32,10 +40,11 @@ function MainDocumentPageToolbar(props: {appModel?: IAppModel}) {
         />
         <i>Double click on the title to Edit</i>
       </div>
-      <button onClick={() => alert("The Add functionality is coming soon")}>Add</button>
-      <button onClick={() => alert("The Save functionality is coming soon")}>Save</button>
-      <button onClick={() => alert("The Load functionality is coming soon")}>Load</button>
-      <button onClick={() => alert("The New functionality is coming soon")}>New</button>
+      <button className="btn" onClick={() => setShowModal(true)}>Add</button>
+      <button className="btn" onClick={() => alert("The Save functionality is coming soon")}>Save</button>
+      <button className="btn" onClick={() => alert("The Load functionality is coming soon")}>Load</button>
+      <button className="btn" onClick={() => alert("The New functionality is coming soon")}>New</button>
+      { showModal && <AddNewCardModal hide={hide}/> }
```

Comment:
It's a very good point.  To be honest, I also don't like this solution but other solutions will look a bit more ugly.

I did it because in our dialog we are creating a new Card and passing it to the state of the component at the creating stage of the component.
After that, we don't destroy this component so the state continues to live with your previous Card even when we hide the Dialog. It wasn't been removed, only hidden so the state is going to be the same.
To solve this issue we can add some conditional logic to our component that will track 
if ->  show dialog -> new Card -> put it to the state
and
if -> hide dialog -> reset the state

also, it could be done with some useEffect callback etc.

so I decided that this solution will be a bit complicated than just mount & unmount a modal dialog. About performance, it's a bit worse than just manipulation via Show\Hide because DOM operations are expensive, but in this particular case it's simpler and it's about Sanity.
So I thought you will like this solution.

If you want me to re-work it the show/hide instead mount/unmount - I'll do it, it isn't a problem
So what do you think?

Reply:
Leave it for now.  I will play with this later.

Reply:
As we changed our component to ClassBased I was able to add a lifecycle hook to control the state behavior. It added additional logic but in this case, it's possible to remove mount/unmount render logic and use show/hide with the correct state.